solr/modules/EnsEMBL/Web/Controller/Ajax.pm

```diff
@@ -353,10 +353,13 @@ sub ajax_psychic { # Invoke psychic via AJAX, to see if we need to redirect.
   $proxy = undef if($SiteDefs::SOLR_NO_PROXY);
   $ua->proxy('http',$proxy) if $proxy;
   $ua->requests_redirectable([]);
-  my $psychic = $SiteDefs::ENSEMBL_PROXY_PROTOCOL.":".
+  my $query_parameter = "q=" . uri_escape($hub->param('q'));
+  my $species_parameter = $hub->param('species') ?  "species=" . uri_escape($hub->param('species')) : undef;
+  my $full_query = join '&', grep { !!$_ } ($query_parameter, $species_parameter);
```

Comment:
i have a feeling you might end up with an extra `&` if there isn't any species_parameter. best to use a simple string concatenation like this:
`$species_parameter ? $query_parameter."&".$species_parameter : $query_parameter`